### PR TITLE
DOC: Replace runtests.py with dev.py where appropriate

### PR DESCRIPTION
**Pull request by melissawm**
This PR removes a few mentions of `runtests.py` in the contributor guide.

cc @tupui

### CODE REVIEW

File: doc/source/dev/contributor/development_workflow.rst [hunk 1]
```diff
@@ -114,10 +114,10 @@ development environment, you'll need to activate your development virtual
 environment, perform an in-place build, and run tests::
 
    conda activate name-of-your-virtual-environment
-   python setup.py build_ext --inplace
-   python runtests.py -v
+   python dev.py build
+   python dev.py test -v
```
Comment (ev-br) [inline doc/source/dev/contributor/development_workflow.rst:118]: Two separate commmands needed? A single `dev.py test` is strictly speaking enough

File: doc/README.md [hunk 2]
```diff
@@ -14,11 +14,11 @@ git submodule update --init
...
 ```
-python3 runtests.py --doc html
+python3 dev.py doc
 ```
 
 Alternatively, if you prefer to build SciPy and the docs separately rather
-than use `runtests.py`:
+than use `dev.py`:
 ```
 python setup.py develop  # in the root of the repo
```
Comment (ev-br) [inline doc/README.md:23]: setup.py develop (while we're at it)

File: doc/source/dev/contributor/ubuntu_guide.rst [hunk 3]
```diff
@@ -204,9 +204,23 @@ With conda
...
 
-#. Do an in-place build: enter ``python3 setup.py build_ext --inplace``. |br| This will compile the C, C++, and Fortran code that comes with SciPy. We installed ``python3`` with Anaconda. ``setup.py`` is a script in the root directory of SciPy, which is why you have to be in the SciPy root directory to call it. ``build_ext`` is a command defined in ``setup.py``, and ``--inplace`` is an option we'll use to ensure that the compiling happens in the SciPy directory you already have rather than the default location for Python packages. By building in-place, you avoid having to re-build SciPy before you can test changes to the Python code.
-
-#. Test the build: enter ``python3 runtests.py -v``. ``runtests.py`` is another script in the SciPy root directory. It runs a suite of tests that make sure SciPy is working as it should, and ``-v`` activates the ``--verbose`` option to show all the test output. If the tests are successful, you now have a working development build of SciPy! You could stop here, but you would only be able to use this development build when the Python working directory is the SciPy root directory.
+#. Build SciPy: enter ``python3 dev.py build`` |br|
```
Comment (ev-br) [inline doc/source/dev/contributor/ubuntu_guide.rst:207]: What is `|br|` and is it needed

**Review by ev-br — COMMENTED** [review]
Looks good modulo small suggestions

File: doc/source/dev/contributor/conda_guide.rst [hunk 4]
```diff
@@ -58,24 +58,23 @@ the latest release of conda from either
...
-     suite of tests that make sure SciPy is working as it should, and ``-v``
-     activates the ``--verbose`` option to show all the test output. If the
-     tests are successful, you now have a working development build of SciPy!
-     You could stop here, but you would only be able to use this development
-     build when the Python working directory is the SciPy root directory.
+#. Build SciPy: enter ``python dev.py build``. 
+
+    This will compile the C, C++, and Fortran code that comes with SciPy and
+    install it in the directory you already have rather than the default
+    location for Python packages. We installed ``python3`` with Anaconda.
```
Comment (tupui) [inline doc/source/dev/contributor/conda_guide.rst:65]:
```suggestion
    location for Python packages. We installed ``python`` with Anaconda.
```

File: doc/source/dev/contributor/quickstart_docker.rst [hunk 5]
```diff
@@ -118,25 +118,24 @@ container do not persist after you close it.
...
-   This will compile the C,
-   C++, and Fortran code that comes with SciPy. ``setup.py`` is a
-   script in the root directory of SciPy, which is why you have to be
-   in the SciPy root directory to call it. ``build_ext`` is a command
-   defined in ``setup.py``, and ``--inplace`` is an option we’ll use to
-   ensure that the compiling happens in the SciPy directory you already
-   have rather than some other folder on your computer. 
+   This will compile the C, C++, and Fortran code that comes with SciPy and
+   install it in the directory you already have rather than the default
+   location for Python packages. We installed ``python3`` with Anaconda.
```
Comment (tupui) [inline doc/source/dev/contributor/quickstart_docker.rst:127]:
```suggestion
   location for Python packages. We installed ``python`` with Anaconda.
```

**Review by tupui — APPROVED** [review]
Thank you @melissawm. Aside from Evgeni's suggestions, just a small thing with python vs python3 to match the command which are used.

While we are editing these, could we use Sphinx-Tabs in these guide as well? (can as well be a follow up if you prefer)

File: doc/README.md (hunk 2 above)
Comment (rgommers) [reply doc/README.md:23]: Maybe just remove this completely, there's no point having two sets of commands and `setup.py develop` is going to go away before the next release anyway.

Comment (melissawm) [reply doc/README.md:23]: Yup I was hesitant on that, but I'm +1 for removing it

File: doc/source/dev/contributor/ubuntu_guide.rst (hunk 3 above)
Comment (melissawm) [reply doc/source/dev/contributor/ubuntu_guide.rst:207]: This is the line break, defined later in the file. Not needed really just a shortcut that was already there, I'm happy to rework and remove :)

**Comment by melissawm** [comment]
@tupui which specific guide did you have in mind? I'd rather do in a follow up but generally yes, happy to do it!

**Comment by tupui** [comment]
> @tupui which specific guide did you have in mind? I'd rather do in a follow up but generally yes, happy to do it!

It was the Ubuntu guide for instance. There is a conda and PyPi section.

**Review by tupui — APPROVED** [review]
Thanks Melissa for the update. Let's get this in and we can follow up for the tabs.

**Pull request merged**